packages/core/src/utils/styleUtils/ssrStyles.ts

```diff
@@ -85,33 +80,20 @@ export const detachExperienceStyles = (experience: Experience): string | undefin
     componentVariablesOverwrites?: Record<string, ComponentPropertyValue>;
     patternWrapper?: ComponentTreeNode;
     wrappingPatternIds: Set<string>;
-    parentChainArr?: string[];
+    wrappingPatternNodeIds?: string[];
   }) => {
     // traversing the tree
-    const queue: QueueItem[] = [];
+    const queue: ComponentTreeNode[] = [];
 
-    queue.push(
-      ...componentTree.children.map((child) => ({
-        node: child,
-        parentChain: [...parentChainArr],
-      })),
-    );
+    queue.push(...componentTree.children);
 
     // for each tree node
     while (queue.length) {
-      const queueItem = queue.shift();
-      if (!queueItem) {
-        break;
-      }
-      const { node: currentNode, parentChain } = queueItem;
-
+      const currentNode = queue.shift();
       if (!currentNode) {
         break;
       }
 
-      const currentNodeParentChain = [...parentChain, currentNode.id || ''];
```

Comment:
Previously, this was a long chain of every single node ID in the hierarchy. Instead of adding the ID on every level we only add those required for the uniqueness check, i.e. prefix the storage key with pattern node IDs only.

Before:
`section1-container1-pattern1-section2-pattern2-section3-container3-heading1`
After:
`pattern1-pattern2-heading1`

Reply:
Has it been tested with the pre-binding scenario?

Currently it relies on the path of ids as breadcrumbs as a way to reach the component from what I understood

We want to change that, but for now let's make sure it works with the current approach without any issues

Reply:
@anwaar931 this won't break anything related to the style propagation for nested patterns, will it?

Reply:
@Chaoste 
<img width="1398" alt="image" src="https://github.com/user-attachments/assets/cacb972f-e398-4bf2-90bf-f26e1131d6a2" />

It looks like there is some issue. We are not generating ssr styles for pattern nodes.

Reply:
<img width="288" alt="image" src="https://github.com/user-attachments/assets/bf612168-a92c-420f-8950-33642daa7cc2" />
This is how my experience looks like